sympy/physics/continuum_mechanics/beam.py

```diff
@@ -2189,13 +2214,64 @@ def draw(self, pictorial=True):
         rectangles += support_rectangles
         markers += support_markers
 
+        if self._composite_type == "hinge":
+            # if self is a composite beam with an hinge, show it
+            ratio = self._hinge_position / self.length
+            x_pos = float(ratio) * length
+            markers += [{'args':[[x_pos], [height / 2]], 'marker':'o', 'markersize':6, 'color':"white"}]
+
+        ylim = (-length, 1.25*length)
+        if fill:
+            # when distributed loads are presents, they might get clipped out
+            # in the figure by the ylim settings.
+            # It might be necessary to compute new limits.
+            _min = min(min(fill["y2"]), min(r["xy"][1] for r in rectangles))
+            _max = max(max(fill["y1"]), max(r["xy"][1] for r in rectangles))
+            if (_min < ylim[0]) or (_max > ylim[1]):
+                offset = abs(_max - _min) * 0.1
+                ylim = (_min - offset, _max + offset)
+
         sing_plot = plot(height + load_eq, height + load_eq1, (x, 0, length),
-         xlim=(-height, length + height), ylim=(-length, 1.25*length), annotations=annotations,
-          markers=markers, rectangles=rectangles, line_color='brown', fill=fill, axis=False, show=False)
+            xlim=(-height, length + height), ylim=ylim,
+            annotations=annotations, markers=markers, rectangles=rectangles,
+            line_color='brown', fill=fill, axis=False, show=False)
 
         return sing_plot
 
 
+    def _is_load_negative(self, load):
+        """Try to determine if a load is negative or positive.
+
+        Returns
+        =======
+        1: if the load is negative
+        0: if the load is positive
+        -1: if it is indeterminate
+
+        Notes
+        =====
+        The Beam class doesn't care about assumptions on symbolic loads.
```

Comment:
The Beam class *should* care about assumptions. That is the only reliable way to know if a symbol or expression involving symbols is positive or negative.

Reply:
It now cares and also tries doit and expand for non-atomic expressions if the give None in their raw state.